### ISSUE TITLE
Examples on DRC don't match current API

### ISSUE THREAD

**Issue by maxlinc** [rackspace/gophercloud#368]
I've been testing samples* based on the snippets on [developer.rackspace.com](https://developer.rackspace.com/docs/cloud-servers/getting-started/?lang=go) and I've noticed as I do this that what I *need* to write to make things work sometimes differs slightly from the snippets.

Two examples:
- A snippet shows `client, err := rackspace.NewIdentityV2(provider)`, but this results in an `assignment count mismatch: 2 = 1` because [NewIdentityV2](http://godoc.org/github.com/rackspace/gophercloud/rackspace#NewIdentityV2) only returns a client.
- Another snippet shows `err := users.Delete(client, "{userId}").ExtractErr()`, but when I try that I get ` undefined (type "github.com/rackspace/gophercloud/openstack/identity/v2/users".DeleteResult has no field or method ExtractErr)`

I wanted to open this issue to note what I'd already found and have a spot to record further discrepancies as I find them, but I'm still adding samples so you may want to mark this as `waiting on feedback` and I'll let you know when I think I have a full list.

**Comment by maxlinc** [comment]
*Note about samples:

I have a tool that allows me to run commands like `crosstest test fog "create server"` or `crosstest test gophercloud "create server"` to test an equivalent "create server" sample in either fog or gophercloud, respectively. If I just run `crosstest test` it will run all the common scenarios for all SDKs. It can perform some cross SDK assertions, comparisons, and produce cross-SDK reports.

I used to just link my scenarios to existing Gophercloud examples that were in this repo, like  `acceptance/04-create-server.go`. At some point those were deleted and replaced with files that are written in a test style rather than an example style - and are harder for me to use. There is now a `acceptance/rackspace/compute/v2/servers_test.go` which has a createServer function, but I'm not sure how (or if) I can run *just* the createServer portion of the test.

So I've stopped using examples from this repo and I'm starting to write my own Gophercloud examples to use for the test suite. I don't think this is a major concern but I wanted to share that the old style of tests were more convenient for me because it was easier for me to identify and isolate examples that matched common scenarios tested by all of the SDKs.
